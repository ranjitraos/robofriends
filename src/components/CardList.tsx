import React from "react";
import Card from "./Card";
import { IRobot } from "./MainPage";

const CardList = ({ robots }: { robots: Array<IRobot> }) => {
    return (
        <div>
            {robots.map(robot => {
                return (
                    <Card
                        key={robot.id}
                        id={robot.id}
                        name={robot.name}
                        email={robot.email}
                    />
                )
            })}
        </div>
    )
}

export default CardList