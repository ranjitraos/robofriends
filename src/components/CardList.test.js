import renderer from 'react-test-renderer';
import CardList from './CardList';

it('renders CardList component correctly', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'jon snow',
            username: 'jon',
            email: 'jon@gmail.com'
        }
    ]
    const cardList = renderer
        .create(<CardList robots={mockRobots} />)
        .toJSON();
    expect(cardList).toMatchSnapshot();
});