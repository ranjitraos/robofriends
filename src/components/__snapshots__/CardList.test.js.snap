// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardList component correctly 1`] = `
<div>
  <div
    className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5 card"
  >
    <img
      alt="robot"
      height={200}
      src="https://robohash.org/1?size=200x200"
      width={200}
    />
    <div>
      <h2>
        jon snow
      </h2>
      <p>
        jon@gmail.com
      </p>
    </div>
  </div>
</div>
`;
